### ISSUE TITLE
Slow Compiling

### ISSUE THREAD

**Issue by latexnoob** [tudace/tuda_latex_templates#60]
Hi,
I am using latex the first time to write my master thesis. I am done with preparing the TUD template and everything seems to work fine. The only thing that bugs me is how much time it takes to compile. It´s at least 20 seconds even tho the template ist still almost empty. My Laptop is pretty powerful and I feel this shouldn´t take that much time.

This is my TexStudio Configuration:
![grafik](https://user-images.githubusercontent.com/56997596/67559268-e97bef00-f718-11e9-905f-682f91df1fad.png)

and this is how I configured the document class:

\documentclass[
	ngerman,
	ruledheaders=section,
	class=report,
	thesis={type=master},
	accentcolor=9c,
	custommargins=geometry,
	marginpar=false,
	%BCOR=5mm,
	parskip=half-,
	fontsize=11pt,
	instbox=false,
	twoside=true
]{tudapub}

\addTitleBoxLogo*{\includegraphics[width=1.0\linewidth]{PTW_Logo_CMYK_whitebackground-eps-converted-to.pdf}}	

I hope there is a way to speed this up.

Thank You!

**Comment by schoeps** [comment]
Latex is not necessarily fast, so it could be that slow. The more packages you the slower it gets.

There may be some ideas to speed up: many editors compile several times (to get all the references right) but this is not always necessary. Maybe this is your case. If so, you can surely turn it off but that requires knowledge on your editor which I do not have. 

However, this question is not related to the design itself. We cannot make latex faster. Sorry, this question is out of scope.

**Comment by TeXhackse** [comment]
Goal is to reduce the compilation frequency. Anyway a full run will take time. 
(For me it's faster than 20 seconds anyway, so maybe your installation is somehow different than mine but this way of debuggin ist totally out of scope)

Your TeXStudio configuration shows, that you probably compile more than once + rerunning Bibliography and other additional programs. Try to compile only once each time. The References might not be right in that moment, but during work on Content that shouldn't matter at all.

The following might not totally fit to your question, thats my copy & paste answer for speed questions:

There are some mechanisms as the draft mode + additionally you could switch to a faster compiler. pdflatex is much faster than xelatex or lualatex. Anyway for a full quality output you will have to wait. 

If you just compile quite often, because you are not sure about the Syntax maybe think about running only a syntax check instead of compiling. Or compile only single chapters of the document.

Images are also something to slow down the process
Additionally you could speed up your installation by caching fonts or create a project specific format file. 

But those things are all about speeding up in general which has nothing to do with the template.
